lib/Core/Executor.cpp

```diff
@@ -1012,24 +1037,7 @@ Executor::regularFork(ExecutionState &current, ref<Expr> condition, bool isInter
   falseState = trueState->branch();
   addedStates.push_back(falseState);
 
-  processTree->attach(current.ptreeNode, falseState, trueState);
-
-  if (pathWriter) {
-    // Need to update the pathOS.id field of falseState, otherwise the same id
-    // is used for both falseState and trueState.
-    falseState->pathOS = pathWriter->open(current.pathOS);
-    if (!isInternal) {
-      trueState->pathOS << "1";
-      falseState->pathOS << "0";
-    }
-  }
-  if (symPathWriter) {
-    falseState->symPathOS = symPathWriter->open(current.symPathOS);
-    if (!isInternal) {
-      trueState->symPathOS << "1";
-      falseState->symPathOS << "0";
-    }
-  }
+  logNewStates(current, trueState, falseState, isInternal);
```

Comment:
Please keep the `attach` out of the log method.

Reply:
Can do, just wonder why?

Reply:
Because they are completely unrelated.

Reply:
`logNewStates` registers new states, `ptree->attach` registers new states in the process tree. I do not think they are unrelated...

Reply:
It doesn't register new states, it just optionally writes path information into files - a rarely used feature - whereas `attach` maintains the process tree.